src/proposalBuilder/executeMinionBuyout.jsx

```diff
@@ -0,0 +1,62 @@
+import React, { useMemo } from 'react';
+import { Box } from '@chakra-ui/react';
+
+import { useDao } from '../contexts/DaoContext';
+import { useInjectedProvider } from '../contexts/InjectedProviderContext';
+import ExecuteSafeMinion from './executeSafeMinion';
+import { ParaSm } from '../components/typography';
+
+const ExecuteMinionBuyout = props => {
+  const { proposal } = props;
+  const { daoMembers } = useDao();
+  const { address } = useInjectedProvider();
+
+  const isMember = useMemo(() => {
+    if (address && daoMembers) {
+      return daoMembers.find(
+        member => member.memberAddress === address && Number(member.shares) > 0,
+      );
+    }
+    return null;
+  }, [address, daoMembers]);
+
+  const memberApplicant = useMemo(() => {
+    if (daoMembers) {
+      const memberMatch = daoMembers.find(
+        member => member.memberAddress === proposal.createdBy,
+      );
+      return {
+        ...memberMatch,
+        canExecute: memberMatch.loot === '0' && memberMatch.shares === '0',
+      };
+    }
+    return null;
+  }, [daoMembers]);
```

Comment:
I'd add address to the dep array so that it updates when the users address changes

Reply:
Well this variable is independent of the user's address. This one only needs to know the member that is the applicant.